document/src/main/java/com/yahoo/document/datatypes/StringFieldValue.java

```diff
@@ -83,60 +187,95 @@ public DataType getDataType() {
+
+        // Deep copy based on current annotation mode
+        switch (getAnnotationMode()) {
+            case SIMPLE:
+                // Deep copy simple annotations
+                copy.simpleAnnotations = new SimpleIndexingAnnotations();
+                for (int i = 0; i < simpleAnnotations.getCount(); i++) {
+                    copy.simpleAnnotations.add(
+                        simpleAnnotations.getFrom(i),
+                        simpleAnnotations.getLength(i),
+                        simpleAnnotations.getTerm(i)
+                    );
+                }
+                copy.spanTrees = null;
+                break;
+
+            case FULL:
+                // Deep copy span trees
+                copy.spanTrees = new HashMap<>(spanTrees.size());
+                for (Map.Entry<String, SpanTree> entry : spanTrees.entrySet()) {
+                    copy.spanTrees.put(entry.getKey(), new SpanTree(entry.getValue()));
+                }
+                copy.simpleAnnotations = null;
+                break;
+
+            case NONE:
+                copy.spanTrees = null;
+                copy.simpleAnnotations = null;
+                break;
         }
-        return strfval;
+
+        return copy;
     }
 
     /** Sets the wrapped String to be an empty String, and clears all span trees. */
     @Override
     public void clear() {
         value = "";
-        if (spanTrees != null) {
-            spanTrees.clear();
-            spanTrees = null;
-        }
+        changeAnnotationMode(AnnotationMode.NONE);
     }
 
     /**
      * Sets a new value for this StringFieldValue. NOTE that doing so will clear all span trees from this value,
      * since they most certainly will not make sense for a new string value.
      *
      * @param o the new String to assign to this. An argument of null is equal to calling clear().
-     * @throws IllegalArgumentException if the given argument is a string containing non-text characters as defined by 
+     * @throws IllegalArgumentException if the given argument is a string containing non-text characters as defined by
      *                                  {@link Text#isTextCharacter(int)}
      */
     @Override
     public void assign(Object o) {
-        if (spanTrees != null) {
-            spanTrees.clear();
-            spanTrees = null;
-        }
+        // Clear existing annotations first
+        changeAnnotationMode(AnnotationMode.NONE);
 
         if (!checkAssign(o)) {
             return;
         }
-        if (o instanceof StringFieldValue) {
-            spanTrees=((StringFieldValue)o).spanTrees;
+
+        if (o instanceof StringFieldValue other) {
+            // Copy only one annotation type based on source's mode
+            switch (other.getAnnotationMode()) {
+                case SIMPLE:
+                    simpleAnnotations = other.simpleAnnotations;
+                    break;
+                case FULL:
+                    spanTrees = other.spanTrees;
```

Comment:
there is no good way to do this, since the annotations should back-ref both this object and the other object